ui/coordinateTrainer/src/ctrl.ts

```diff
@@ -2,11 +2,33 @@ import { sparkline } from '@fnando/sparkline';
 import * as xhr from 'common/xhr';
 import throttle from 'common/throttle';
 import { Api as CgApi } from 'chessground/api';
-import { ColorChoice, CoordinateTrainerConfig, InputMethod, Mode, ModeScores, Redraw } from './interfaces';
+import {
+  ColorChoice,
+  TimeControlChoice,
+  CoordinateTrainerConfig,
+  InputMethod,
+  Mode,
+  ModeScores,
+  Redraw,
+} from './interfaces';
 
 const orientationFromColorChoice = (colorChoice: ColorChoice): Color =>
   (colorChoice === 'random' ? ['white', 'black'][Math.round(Math.random())] : colorChoice) as Color;
 
+const durationFromTimeControlChoice = (timeControlChoice: TimeControlChoice): number => {
+  switch (timeControlChoice) {
+    case 'thirtySeconds':
+      return 30 * 1000;
+    case 'oneMinute':
+      return 60 * 1000;
+    case 'noTime':
+      return 0;
+    default:
```

Comment:
Nitpick, but do we need a default here? If the type is to be believed, it should never be triggered, correct?